dev/benchmarks/macrobenchmarks/lib/web_benchmarks.dart

```diff
@@ -50,7 +55,36 @@ final Map<String, RecorderFactory> benchmarks = <String, RecorderFactory>{
     BenchTextCachedLayout.canvasBenchmarkName: () => BenchTextCachedLayout(useCanvas: true),
     BenchBuildColorsGrid.domBenchmarkName: () => BenchBuildColorsGrid.dom(),
     BenchBuildColorsGrid.canvasBenchmarkName: () => BenchBuildColorsGrid.canvas(),
-  }
+  },
+
+  // The following benchmarks are for the Flutter Gallery.
+  '${_galleryBenchmarkPrefix}_studies_perf': () => GalleryRecorder(
+    benchmarkName: '${_galleryBenchmarkPrefix}_studies_perf',
+    reporter: _client.reportDemo,
+    reportError: _client.reportError,
+    runCriterion: (String demo) => typeOfDemo(demo) == DemoType.study,
+  ),
+  '${_galleryBenchmarkPrefix}_animated_widgets_perf': () => GalleryRecorder(
+    benchmarkName: '${_galleryBenchmarkPrefix}_animated_widgets_perf',
+    reporter: _client.reportDemo,
+    reportError: _client.reportError,
+    runCriterion: (String demo) =>
+    typeOfDemo(demo) == DemoType.animatedWidget,
+  ),
+  '${_galleryBenchmarkPrefix}_unanimated_widgets_perf': () => GalleryRecorder(
+    benchmarkName: '${_galleryBenchmarkPrefix}_unanimated_widgets_perf',
+    reporter: _client.reportDemo,
+    reportError: _client.reportError,
+    runCriterion: (String demo) =>
+    typeOfDemo(demo) == DemoType.unanimatedWidget,
+  ),
+  '${_galleryBenchmarkPrefix}_scroll_perf': () => GalleryRecorder(
+    benchmarkName: '${_galleryBenchmarkPrefix}_scroll_perf',
+    reporter: _client.reportDemo,
+    reportError: _client.reportError,
+    runCriterion: (String demo) => false,
```

Comment:
Should `runCriterion` be necessary if the benchmark `testScrollsOnly`?

Reply:
I can make `runCriterion` (`runPredicate`) optional, so it can be left as `null` in this case.

Reply:
SGTM

Reply:
Done.